test/errors/throw-error-with-getter-throw-traced.mjs

```diff
@@ -0,0 +1,27 @@
+import { spawnPromisified } from '../common/index.mjs';
+import assert from 'node:assert';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+
+
+describe('--trace-uncaught', () => {
+  it('prints a trace on process exit for uncaught errors', async () => {
+    const { code, signal, stderr } = await spawnPromisified(execPath, [
+      '--trace-uncaught',
+      '--eval',
+      `throw {
+        get stack() {
+          throw new Error('weird throw but ok');
+        },
+        get name() {
+          throw new Error('weird throw but ok');
+        },
+      };`,
+    ]);
+
+    assert.match(stderr, /Thrown at:/);
```

Comment:
I think this test is now testing less than it was before, why not keeping it as a `message/` test like before?
```suggestion
    assert.match(stderr, /^Thrown at:$/m);
```

Reply:
See the explanation in the PR description.

Reply:
An alternative would be to replace CJS-related stuff with `*`, e.g.:

```

*:4
throw {  // eslint-disable-line no-throw-literal
^
[object Object]
Thrown at:
    at *throw_error_with_getter_throw_traced.js:*:*
    at *
    at *
    at *
    at *
    at *
    at *

Node.js *
```

Reply:
Then the test breaks as soon as the number of calls changes, since the trace length is less than 10.

I really don't think keeping this in the .out style is important; the test is meant to validate that a trace is printed, not the format of the trace.

Reply:
The test is called `throw_error_with_getter_throw_traced`, and if you look at the .js file it's clear that the test is checking that a `throw` of an object with a getter causes a trace under `--trace-uncaught`.

All the other .out files with traces are testing the format of the trace presentation.

Reply:
Stack trace of regular `Error` objects and stack trace generated by `--trace-uncaught` are not the same thing (that's also why `Error.stackTraceLimit` doesn't apply to those stack traces). So I don't think it's fair to say that it's validated by the other tests that would check the format of regular `Error` object stack traces. I think it's worth validating the whole output, doesn't have to be in a `.out` file, it could be a big regex (but maybe `.out` file is easier)

Reply:
Is it possible to use `.out` and support a variable length stack trace? It seems like `*` always matches within a line, not multiline.

Some of the `.out` files contain lines like `    [... lines matching original stack trace ...]` which I have no idea what this means, and I can’t find any documentation about. Is this a Python regex thing?

Reply:
> Some of the .out files contain lines like [... lines matching original stack trace ...]

This comes from `util.inspect` on some errors, I believe.

Reply:
> This comes from `util.inspect` on some errors, I believe.

As in, the actual output contains that text, and it’s not just some weird regex replacement?

Reply:
Looks like I was wrong. It's specific to errors from event emitters: https://github.com/nodejs/node/blob/37115034dded06f54876a3b4784c5c478e8b65e1/lib/events.js#L447

Reply:
But yeah, the actual stack trace value contains that text.

Reply:
Okay, so then we’re back to the original issue, which is that I don’t know of any way to have `*` (or something similar) match multiple lines in an `.out` file. If there’s no way to do a multiline match, we can’t use `.out` files to test for varying-length stack traces, which is a requirement for what I’m trying to achieve here.

I can change the new tests to test for more things in the output or use a huge regex that’s essentially a representation of what the `.out` file gets parsed into by the Python script runner; the question there is really what is our intention? As in, for the test `throw-error-with-getter-throw-traced.mjs` what in the output are we really trying to validate? My impression is that we’re checking for the _existence_ of a stack trace, but not anything about its presentation. Regardless of the original intent when the test was written, what do we really want to validate for this feature today?